core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -164,6 +164,27 @@ private[spark] object TestUtils {
     createCompiledClass(className, destDir, sourceFile, classpathUrls)
   }
 
+  /** Create a dummy compile jar for a given package, classname.  Jar will be placed in destDir */
+  def createDummyJar(destDir: String, packageName: String, className: String): String = {
```

Comment:
The R tests do indeed verify that they can call the internal functions.

I can revert that part of the changes.